datafusion/sql/tests/sql_integration.rs

```diff
@@ -55,6 +55,35 @@ use sqlparser::dialect::{Dialect, GenericDialect, HiveDialect, MySqlDialect};
 mod cases;
 mod common;
 
+pub struct ParameterTest<'a> {
+    pub sql: &'a str,
+    pub expected_types: Vec<(&'a str, Option<DataType>)>,
+    pub param_values: Vec<ScalarValue>,
+}
+
+#[allow(clippy::needless_lifetimes)]
+impl<'a> ParameterTest<'a> {
```

Comment:
I think you can avoid the clippy issue with something like this:

```suggestion
impl ParameterTest<'_> {
```